hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestLeaseRecovery.java

```diff
@@ -433,6 +433,27 @@ public void testRecoverWrongFile() throws Exception {
     }
   }
 
+  @Test
+  public void testRecoveryWithoutBlocks() throws Exception {
+    RootedOzoneFileSystem fs = (RootedOzoneFileSystem)FileSystem.get(conf);
+
+    final FSDataOutputStream stream = fs.create(file, true);
+    try {
+      stream.hsync();
+      assertFalse(fs.isFileClosed(file));
+
+      int count = 0;
+      while (count++ < 15 && !fs.recoverLease(file)) {
+        Thread.sleep(1000);
+      }
+      // The lease should have been recovered.
+      assertTrue(fs.isFileClosed(file), "File should be closed");
+
+    } finally {
+      closeIgnoringKeyNotFound(stream);
```

Comment:
is it supposed to ignore if key is not found? can we just do stream.close() instead

Reply:
I see other tests in this class follows the same. After lease recovery it throws KEY_NOT_FOUND if we close the stream.

Reply:
Ok let's revisit later.